scipy/spatial/qhull.pyx

```diff
@@ -2410,3 +2449,269 @@ class Voronoi(_QhullUser):
             self._ridge_dict = dict(zip(map(tuple, self.ridge_points.tolist()),
                                         self.ridge_vertices))
         return self._ridge_dict
+
+#------------------------------------------------------------------------------
+# Halfspace intersection interface, for Python
+#------------------------------------------------------------------------------
+
+class _QhalfUser(object):
```

Comment:
So, after some testing, the incremental mode works. So I will now re-merge QhalfUser into QhullUser. Note that two methods are changed:
- _update(qhull) will have to become _update_halfspaces() and _update_points()
- add_points() will become _add_points() and _add_halfspaces()